packages/lodestar/src/chain/chain.ts

```diff
@@ -176,27 +168,46 @@ export class BeaconChain extends (EventEmitter as {new (): ChainEventEmitter}) i
     epochCtx.loadState(state);
     await this.db.stateCache.add({state, epochCtx});
     this.logger.info("Chain started, waiting blocks and attestations");
-    this.clock = new LocalClock(this.config, state.genesisTime);
-    await this.clock.start();
-    await this.forkChoice.start(state.genesisTime, this.clock);
+    this.clock = new LocalClock({
+      config: this.config,
+      emitter: this.emitter,
+      genesisTime: state.genesisTime,
+      signal: this.abortController.signal,
+    });
+    this.forkChoice = new ArrayDagLMDGHOST({
+      config: this.config,
+      emitter: this.emitter,
+      genesisTime: state.genesisTime,
+    });
+    this.attestationProcessor = new AttestationProcessor(this, {
+      config: this.config,
+      db: this.db,
+      logger: this.logger,
+    });
+    this.blockProcessor = new BlockProcessor(
+      this.config,
+      this.logger,
+      this.db,
+      this.forkChoice,
+      this.metrics,
+      this.emitter,
+      this.attestationProcessor
+    );
```

Comment:
Why constructing in start method?

Reply:
fork choice, clock, and processors now require an anchor state to be constructed, allows us to make them all simpler.